game/src/core/systems/CameraSystem.java

```diff
@@ -79,8 +79,9 @@ public static OrthographicCamera camera() {
 
   @Override
   public void execute() {
-    if (entityStream().findAny().isEmpty()) focus();
-    else entityStream().forEach(this::focus);
+    if (filteredEntityStream(CameraComponent.class, PositionComponent.class).findAny().isEmpty())
+      focus();
+    else filteredEntityStream().forEach(this::focus);
```

Comment:
```suggestion
    filteredEntityStream(CameraComponent.class, PositionComponent.class)
        .findAny()
        .ifPresentOrElse(this::focus, this::focus);
```

Reply:
@AMatutat Wenn der Stream leer ist (also niemand eine `CameraComponent` hat), dann soll einfach die parameterlose Methode `CameraSystem#focus` aufgerufen werden. Anderenfalls für jedes Element einmal die Überladung mit der Entität. 

Ich sehe hier zwei Probleme: 

1. Wird hier bis zu zweimal der selbe Stream geholt.
2. Wenn es mehrere Entitäten mit `CameraComponent` gibt, wird für jede einmal `CameraSystem#focus` aufgerufen?! Dabei überschreiben sich die Aufrufe, und nur der letzte ist wirksam?

Wenn die Diagnose korrekt ist, würde ich `ifPresentOrElse` vorschlagen. Falls es Elemente mit `CameraComponent` gibt, wird für eines davon `focus(e)` aufgerufen und `focus()` sonst. Macht das Sinn?

Reply:
PS: Wenn es jetzt nicht gleich drei `focus`-Methoden in der Klasse geben würde, dann könnte man den `ifPresentOrElse `-Ausdruck sogar lesen :)  .... (also wenn eine `focusCurrentPosition`, eine `focusEntity` und eine `focusPoint` o.ä. heissen würde)

Reply:
> Wenn es mehrere Entitäten mit CameraComponent gibt, wird für jede einmal CameraSystem#focus aufgerufen?! Dabei überschreiben sich die Aufrufe, und nur der letzte ist wirksam?

Mit `findAny` müsste es doch der erste sein, oder? Anyway ist eh ein undefined behavior: 
Aus CameraComponent 

> Note: If there is more than one CameraComponent, i.e. if more than one entity is attached to a
 * CameraComponent, the behaviour is undefined.


> PS: Wenn es jetzt nicht gleich drei focus-Methoden in der Klasse geben würde, dann könnte man den ifPresentOrElse -Ausdruck sogar lesen :) .... (also wenn eine focusCurrentPosition, eine focusEntity und eine focusPoint o.ä. heissen würde)

Finde das auch so lesbar, vielleicht sogar lesbarer da überladen von Funktionen immer ähnliches verhalten impliziert.

Außerdem endet eh alles in
```java
  private void focus(Point point) {
    CAMERA.position.set(point.x, point.y, 0);
  }
```


Außerdem: Eigentlich nix für diesen PR

Reply:
> Mit findAny müsste es doch der erste sein, oder?

Nö: Es findet _irgendein_ Element, lt. Javadoc sogar hochgradig nichtdeterministisch. Das erste wäre `findFirst`.

> Außerdem endet eh alles in `private void focus(Point point)`

Richtig. Deshalb macht die `forEach`-Schleife für mich auch irgendwie so gar keinen Sinn - nur der letzte Aufruf "überlebt". Das kann man dann auch vernünftig hinschreiben und halt sagen: nimm irgendeinen (falls es mehrere gibt).

> Außerdem: Eigentlich nix für diesen PR

Im Prinzip ja, aber da hier eh ein erweitertes Refactoring (inkl. Änderung der API!) gemacht wird, würde ich das hier gleich mit durchziehen.